nncf/openvino/engine.py

```diff
@@ -72,11 +72,18 @@ def __iter__(self):
 
     def __len__(self) -> int:
         if self._length is None:
-            self._length = DatasetWrapper._get_length(self._iterable)
+            indices = getattr(self._iterable, '_indices', None)
+            data_source = getattr(self._iterable, '_data_source', None)
```

Comment:
Is there a way for this be rewritten in terms of `isinstance` calls? `isinstance` is bad on its own, though, but plain `getattr`-based conditionals are even worse.